ably/realtime/connection.py

```diff
@@ -152,24 +165,26 @@ def __init__(self, realtime, initial_state):
         self.__retry_task = None
         self.__connection_details = None
         self.__fail_state = ConnectionState.DISCONNECTED
+        self.__fail_event = ConnectionEvent.DISCONNECTED
         super().__init__()
 
-    def enact_state_change(self, state, reason=None):
+    def enact_state_change(self, state, event, reason=None):
```

Comment:
I think this signature for `enact_state_change` looks a bit strange and `Connection` will still emit the state as the event. I think it would be better to just emit an `'update'` instead of a `'connectionstate'` for when we recieve a `CONNECTED` message while connected, and then add a handler for this in the `Connection` constructor (separate from the one for `'connectionstate').